### PR TITLE
UICAL-148 use react-intl for date formatting

### PR DESCRIPTION
**Pull request by zburke**
## Purpose

Some date display was pretty convoluted, storing formatters as translations and passing those to moment and passing that to `dateformat` and ... yeesh. 

## Approach

Use `react-intl` externally to present values to the UI, i.e. don't call `moment` functions for display. This resolves date strings like `01/20/AMAMAMAM` which passed the `dateFormat` translation value as an argument to `moment.format()`. This also allows the `dateformat` dependency to be removed completely. 

Additionally, use the localized value for `Closed`. It was always available, but we just weren't using it.

Finally, add `react-hot-loader` as a dev-dep and update `index.js` to support hot-loading. This is a standard practice for other modules that was missing here, but arguably that should be handled in a separate PR; LMK if you would prefer that. I didn't actually intend to include it here but forgot to split the changes on `package.json`. 

Refs [UICAL-148](https://issues.folio.org/browse/UICAL-148)

### CODE REVIEW

File: translations/ui-calendar/en.json [hunk 1]
```diff
@@ -69,7 +69,6 @@
   "regularLibraryHoursCalendar":"Regular Library Hours:",
   "selectTemplate":"Select Template",
   "copy":"Copy Previous",
-  "dateFormat":"MM/DD/YYYY",
```
Comment (Dmitriy-Litvinenko) [inline translations/ui-calendar/en.json:72]: Used in ui-calendar\stripes\ui-calendar\test\bigtest\helpers\messageConverters.js
We should change formatDateString before deleting dateFormat
Just for example `return date.format('MM/DD/YYYY');`

File: src/settings/ServicePointDetails.js [hunk 2]
```diff
@@ -207,7 +197,13 @@ class ServicePointDetails extends React.Component {
                   xs={11}
                   data-test-next-period-item-label
                 >
-                  {this.state.currentPeriod.startDate + ' - ' + this.state.currentPeriod.endDate + ' (' + this.state.currentPeriod.name + ')'}
+                  {this.state.currentPeriod.startDate}
```
Comment (Dmitriy-Litvinenko) [inline src/settings/ServicePointDetails.js:200]: We have test error here "AssertionError: expected '7/25/2021-8/29/2021 (test)' to equal '07/25/2021 - 08/29/2021 (test)'"

**Review by Dmitriy-Litvinenko — CHANGES_REQUESTED** [review]

**Comment by Dmitriy-Litvinenko** [comment]
@zburke hello thank you for fix, but unfortunately we have problems with tests. I found 2 problems and add comment to them.

**Comment by zburke** [comment]
Yeah, I should have let the tests run on the branch before tagging you for review, @Dmitriy-Litvinenko. Thanks for looking into this! I will try to get it cleaned up with minimal effort.

**Review by Dmitriy-Litvinenko — APPROVED** [review]

**Pull request merged**